app/src/common/shared/com/igalia/wolvic/ui/widgets/WindowWidget.java

```diff
@@ -2099,10 +2099,10 @@ public void onShowActionRequest(@NonNull WSession aSession, @NonNull Selection a
         RectF selectionRect = null;
         RectF clientRect = aSelection.clientRect();
         if (clientRect != null) {
-            float ratio = WidgetPlacement.worldToWindowRatio(getContext());
+            float ratio = WidgetPlacement.worldToWindowRatio(getContext()) / mWidgetPlacement.textureScale / SettingsStore.getInstance(getContext()).getDisplayDensity();
```

Comment:
What about doing this in worldToWindowRatio (this is the only caller AFAIK) with a comment explaining why it's needed?

That way, future callers wouldn't need to apply the same transformation.

Reply:
And BTW it'd be nice if you could add a more verbose description in the commit. Basically I'd include a description of what happens (menu is invisible), what caused it (as you do already) and how it was fixed.

Reply:
Updated, thanks for pointing that out.